spec-tests/src/test/java/org/candlepin/spec/pools/RefreshPoolsTest.java

```diff
@@ -0,0 +1,883 @@
+        }
+        else {
+            assertNull(status.getResultData());
+        }
+
+        OwnerDTO owner = adminClient.owners().getOwner(expectedKey);
+        assertNotNull(owner);
+        assertEquals(expectedKey, owner.getDisplayName());
+        assertEquals(Owners.SCA_ACCESS_MODE, owner.getContentAccessMode());
+    }
+
+    @Test
+    @OnlyInHosted
+    public void shouldCreateTheCorrectNumberOfPools() {
+        int expectedSubscriptionCount = 6;
+        Set<String> expectedSubIds = new HashSet<>();
+        for (int i = 0; i < expectedSubscriptionCount; i++) {
+            ProductDTO prod = adminClient.hosted().createProduct(Products.random());
+            SubscriptionDTO sub = adminClient.hosted().createSubscription(Subscriptions.random(owner, prod));
+            expectedSubIds.add(sub.getId());
+        }
+
+        refreshPools(adminClient, ownerKey);
+
+        List<PoolDTO> pools = adminClient.pools().listPoolsByOwner(owner.getId());
+        assertThat(pools)
+            .hasSize(expectedSubscriptionCount)
+            .map(PoolDTO::getSubscriptionId)
+            .isNotNull()
+            .containsExactlyInAnyOrderElementsOf(expectedSubIds);
+    }
+
+    @Test
+    @OnlyInHosted
+    public void shouldDetectChangesInProviderProducts() {
+        ProductDTO prod1 = adminClient.hosted().createProduct(Products.random());
+        ProductDTO prod2 = adminClient.hosted().createProduct(Products.random());
+        ProductDTO providerProduct = Products.random();
+        providerProduct.setProvidedProducts(Set.of(prod1, prod2));
+        providerProduct = adminClient.hosted().createProduct(providerProduct);
+        adminClient.hosted().createSubscription(Subscriptions.random(owner, providerProduct));
+
+        refreshPools(adminClient, ownerKey);
+        List<PoolDTO> pools = adminClient.pools().listPoolsByOwner(owner.getId());
+        assertEquals(1, pools.size());
+
+        PoolDTO actualPool = pools.get(0);
+        Set<String> expectedProvProdIds = Set.of(prod1.getId(), prod2.getId());
+        assertThat(actualPool.getProvidedProducts())
+            .isNotNull()
+            .map(ProvidedProductDTO::getProductId)
+            .hasSameSizeAs(expectedProvProdIds)
+            .hasSameElementsAs(expectedProvProdIds);
+
+        // Remove the old provided products and add a new one...
+        ProductDTO prod3 = adminClient.hosted().createProduct(Products.random());
+        providerProduct.setProvidedProducts(Set.of(prod3));
+        providerProduct = adminClient.hosted().updateProduct(providerProduct.getId(), providerProduct);
+
+        refreshPools(adminClient, ownerKey);
+        pools = adminClient.pools().listPoolsByOwner(owner.getId());
+        assertThat(pools)
+            .isNotNull()
+            .hasSize(1);
+
+        assertThat(pools.get(0).getProvidedProducts())
+            .isNotNull()
+            .singleElement()
+            .returns(prod3.getId(), from(ProvidedProductDTO::getProductId));
```

Comment:
Not a change request, but just making sure you know: the `from` here is purely optional syntactic sugar to make it "read" like an English sentence.